daemon/daemon.go

```diff
@@ -922,12 +922,17 @@ func (daemon *Daemon) GetRemappedUIDGID() (int, int) {
 	return uid, gid
 }
 
-// tempDir returns the default directory to use for temporary files.
-func tempDir(rootDir string, rootUID, rootGID int) (string, error) {
+// prepareTempDir prepares and returns the default directory to use
+// for temporary files.
+// If it doesn't exist, it is created. If it exists, its content is removed.
+func prepareTempDir(rootDir string, rootUID, rootGID int) (string, error) {
 	var tmpDir string
 	if tmpDir = os.Getenv("DOCKER_TMPDIR"); tmpDir == "" {
 		tmpDir = filepath.Join(rootDir, "tmp")
+		os.RemoveAll(tmpDir)
```

Comment:
We should have the Docker for Mac team have a look as well; IIRC they store other data in this directory, and I don't know if that can be removed; IIRC, we use specific prefixes for partial pulls, and build contexts, so we could be more specific and just remove those directories

In addition, we recently merged a PR for scenarios where docker when running dind (see https://github.com/docker/docker/pull/31265); which also removed the _content_ instead of the directory itself would that make sense?